### ISSUE TITLE
problem for running the latest image (3 ways)

### ISSUE THREAD

**Issue by MdeLv** [sickcodes/Docker-OSX#100]
I was unable to get started with the latest image:
- from a DSM synology which has docker integrated (the container stopped in an unexpected way ; no more details)
- from a debian10 VM / docker hosted on the DSM
- from a debian 10 /docker on a physical motherboard (on going test ; not enough space in var/ ...)

From the VM debian10, here are the 2 trials (I'm not fully aware if the commands are correct) 
Can you give some hints to get this image up&running?
Thanks.

```
$ sudo docker run --device /dev/kvm --device /dev/snd -v /tmp/.X11-unix:/tmp/.X11-unix -e "DISPLAY=${DISPLAY:-:0.0}"     sickcodes/docker-osx:latest
docker: Error response from daemon: error gathering device information while adding custom device "/dev/kvm": no such file or directory.
ERRO[0003] error waiting for container: context canceled 

$ sudo docker run --privileged -e "DISPLAY=${DISPLAY:-:0.0}" -v /tmp/.X11-unix:/tmp/.X11-unix sickcodes/docker-osx
ssh-keygen: generating new host keys: RSA DSA ECDSA ED25519 
No protocol specified
Unable to init server: Could not connect: Connection refused
Could not access KVM kernel module: No such file or directory
qemu-system-x86_64: failed to initialize kvm: No such file or directory
qemu-system-x86_64: falling back to tcg
QEMU 5.1.0 monitor - type 'help' for more information
(qemu) qemu-system-x86_64: cannot set up guest memory 'pc.ram': Cannot allocate memory
```

**Comment by sickcodes** [comment]
KVM disallowed, meaning hardware virtualization likely disabled. Running this inside a guest means you'll be 3x nested deep.

Host -> Debian -> Docker (Arch btw) -> OSX

**Comment by MdeLv** [comment]
Regarding docken on virtualized debian, is there a way to emulate hardware virtualization? (forget about the performance issue, just need to run MacOS some softwares).

Regarding synology DSM with docker integrated, here is the log I could get.
Is there a way to workaround the issues such as "qemu-system-x86_64: warning: TCG doesn't support requested feature" etc. ?

Thanks.

```
sickcodes-docker-osx1
date	stream	content
2020-11-02 17:58:21	stdout	gtk initialization failed
2020-11-02 17:58:21	stdout	audio: Failed to create voice `adc'
2020-11-02 17:58:21	stdout	alsa: Reason: No such file or directory
2020-11-02 17:58:21	stdout	alsa: Failed to open `default':
2020-11-02 17:58:21	stdout	alsa: Could not initialize ADC
2020-11-02 17:58:21	stdout	ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM default
2020-11-02 17:58:21	stdout	ALSA lib conf.c:5233:(snd_config_expand) Evaluate error: No such file or directory
2020-11-02 17:58:21	stdout	ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
2020-11-02 17:58:21	stdout	ALSA lib confmisc.c:1246:(snd_func_refer) error evaluating name
2020-11-02 17:58:21	stdout	ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory
2020-11-02 17:58:21	stdout	ALSA lib confmisc.c:392:(snd_func_concat) error evaluating strings
2020-11-02 17:58:21	stdout	ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_card_driver returned error: No such file or directory
2020-11-02 17:58:21	stdout	ALSA lib confmisc.c:767:(parse_card) cannot find card '0'
2020-11-02 17:58:21	stdout	alsa: Reason: No such file or directory
2020-11-02 17:58:21	stdout	alsa: Failed to open `default':
2020-11-02 17:58:21	stdout	alsa: Could not initialize ADC
2020-11-02 17:58:21	stdout	ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM default
2020-11-02 17:58:21	stdout	ALSA lib conf.c:5233:(snd_config_expand) Evaluate error: No such file or directory
2020-11-02 17:58:21	stdout	ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
2020-11-02 17:58:21	stdout	ALSA lib confmisc.c:1246:(snd_func_refer) error evaluating name
2020-11-02 17:58:21	stdout	ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory
2020-11-02 17:58:21	stdout	ALSA lib confmisc.c:392:(snd_func_concat) error evaluating strings
2020-11-02 17:58:21	stdout	ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_card_driver returned error: No such file or directory
2020-11-02 17:58:21	stdout	ALSA lib confmisc.c:767:(parse_card) cannot find card '0'
2020-11-02 17:58:21	stdout	audio: Failed to create voice `dac'
2020-11-02 17:58:21	stdout	alsa: Reason: No such file or directory
2020-11-02 17:58:21	stdout	alsa: Failed to open `default':
2020-11-02 17:58:21	stdout	alsa: Could not initialize DAC
2020-11-02 17:58:21	stdout	ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM default
2020-11-02 17:58:21	stdout	ALSA lib conf.c:5233:(snd_config_expand) Evaluate error: No such file or directory
2020-11-02 17:58:21	stdout	ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
2020-11-02 17:58:21	stdout	ALSA lib confmisc.c:1246:(snd_func_refer) error evaluating name
2020-11-02 17:58:21	stdout	ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory
2020-11-02 17:58:21	stdout	ALSA lib confmisc.c:392:(snd_func_concat) error evaluating strings
2020-11-02 17:58:21	stdout	ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_card_driver returned error: No such file or directory
2020-11-02 17:58:21	stdout	ALSA lib confmisc.c:767:(parse_card) cannot find card '0'
2020-11-02 17:58:21	stdout	alsa: Reason: No such file or directory
2020-11-02 17:58:21	stdout	alsa: Failed to open `default':
2020-11-02 17:58:21	stdout	alsa: Could not initialize DAC
2020-11-02 17:58:21	stdout	ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM default
2020-11-02 17:58:21	stdout	ALSA lib conf.c:5233:(snd_config_expand) Evaluate error: No such file or directory
2020-11-02 17:58:21	stdout	ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
2020-11-02 17:58:21	stdout	ALSA lib confmisc.c:1246:(snd_func_refer) error evaluating name
2020-11-02 17:58:21	stdout	ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory
2020-11-02 17:58:21	stdout	ALSA lib confmisc.c:392:(snd_func_concat) error evaluating strings
2020-11-02 17:58:21	stdout	ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_card_driver returned error: No such file or directory
2020-11-02 17:58:21	stdout	ALSA lib confmisc.c:767:(parse_card) cannot find card '0'
2020-11-02 17:58:21	stdout	qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.80000007H:EDX.invtsc [bit 8]
2020-11-02 17:58:21	stdout	qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.01H:ECX.avx [bit 28]
2020-11-02 17:58:21	stdout	qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.01H:ECX.pcid [bit 17]
2020-11-02 17:58:21	stdout	qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.80000007H:EDX.invtsc [bit 8]
2020-11-02 17:58:21	stdout	qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.01H:ECX.avx [bit 28]
2020-11-02 17:58:21	stdout	qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.01H:ECX.pcid [bit 17]
2020-11-02 17:58:21	stdout	qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.80000007H:EDX.invtsc [bit 8]
2020-11-02 17:58:21	stdout	qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.01H:ECX.avx [bit 28]
2020-11-02 17:58:21	stdout	qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.01H:ECX.pcid [bit 17]
2020-11-02 17:58:21	stdout	qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.80000007H:EDX.invtsc [bit 8]
2020-11-02 17:58:21	stdout	qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.01H:ECX.avx [bit 28]
2020-11-02 17:58:21	stdout	(qemu) qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.01H:ECX.pcid [bit 17]
2020-11-02 17:58:21	stdout	QEMU 5.1.0 monitor - type 'help' for more information
2020-11-02 17:58:21	stdout	qemu-system-x86_64: falling back to tcg
2020-11-02 17:58:21	stdout	qemu-system-x86_64: failed to initialize kvm: No such file or directory
2020-11-02 17:58:21	stdout	Could not access KVM kernel module: No such file or directory
2020-11-02 17:58:21	stdout	Unable to init server: Could not connect: Connection refused
2020-11-02 17:58:20	stdout	nohup: appending output to 'nohup.out'
2020-11-02 17:57:29	stdout	gtk initialization failed
2020-11-02 17:57:29	stdout	audio: Failed to create voice `adc'
2020-11-02 17:57:29	stdout	alsa: Reason: No such file or directory
2020-11-02 17:57:29	stdout	alsa: Failed to open `default':
2020-11-02 17:57:29	stdout	alsa: Could not initialize ADC
2020-11-02 17:57:29	stdout	ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM default
2020-11-02 17:57:29	stdout	ALSA lib conf.c:5233:(snd_config_expand) Evaluate error: No such file or directory
2020-11-02 17:57:29	stdout	ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
2020-11-02 17:57:29	stdout	ALSA lib confmisc.c:1246:(snd_func_refer) error evaluating name
2020-11-02 17:57:29	stdout	ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory
2020-11-02 17:57:29	stdout	ALSA lib confmisc.c:392:(snd_func_concat) error evaluating strings
2020-11-02 17:57:29	stdout	ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_card_driver returned error: No such file or directory
2020-11-02 17:57:29	stdout	ALSA lib confmisc.c:767:(parse_card) cannot find card '0'
2020-11-02 17:57:29	stdout	alsa: Reason: No such file or directory
2020-11-02 17:57:29	stdout	alsa: Failed to open `default':
2020-11-02 17:57:29	stdout	alsa: Could not initialize ADC
2020-11-02 17:57:29	stdout	ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM default
2020-11-02 17:57:29	stdout	ALSA lib conf.c:5233:(snd_config_expand) Evaluate error: No such file or directory
2020-11-02 17:57:29	stdout	ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
2020-11-02 17:57:29	stdout	ALSA lib confmisc.c:1246:(snd_func_refer) error evaluating name
2020-11-02 17:57:29	stdout	ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory
2020-11-02 17:57:29	stdout	ALSA lib confmisc.c:392:(snd_func_concat) error evaluating strings
2020-11-02 17:57:29	stdout	ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_card_driver returned error: No such file or directory
2020-11-02 17:57:29	stdout	ALSA lib confmisc.c:767:(parse_card) cannot find card '0'
2020-11-02 17:57:29	stdout	audio: Failed to create voice `dac'
2020-11-02 17:57:29	stdout	alsa: Reason: No such file or directory
2020-11-02 17:57:29	stdout	alsa: Failed to open `default':
2020-11-02 17:57:29	stdout	alsa: Could not initialize DAC
2020-11-02 17:57:29	stdout	ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM default
2020-11-02 17:57:29	stdout	ALSA lib conf.c:5233:(snd_config_expand) Evaluate error: No such file or directory
2020-11-02 17:57:29	stdout	ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
2020-11-02 17:57:29	stdout	ALSA lib confmisc.c:1246:(snd_func_refer) error evaluating name
2020-11-02 17:57:29	stdout	ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory
2020-11-02 17:57:29	stdout	ALSA lib confmisc.c:392:(snd_func_concat) error evaluating strings
2020-11-02 17:57:29	stdout	ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_card_driver returned error: No such file or directory
2020-11-02 17:57:29	stdout	ALSA lib confmisc.c:767:(parse_card) cannot find card '0'
2020-11-02 17:57:29	stdout	alsa: Reason: No such file or directory
2020-11-02 17:57:29	stdout	alsa: Failed to open `default':
2020-11-02 17:57:29	stdout	alsa: Could not initialize DAC
2020-11-02 17:57:29	stdout	ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM default
2020-11-02 17:57:29	stdout	ALSA lib conf.c:5233:(snd_config_expand) Evaluate error: No such file or directory
2020-11-02 17:57:29	stdout	ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
2020-11-02 17:57:29	stdout	ALSA lib confmisc.c:1246:(snd_func_refer) error evaluating name
2020-11-02 17:57:29	stdout	ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory
2020-11-02 17:57:29	stdout	ALSA lib confmisc.c:392:(snd_func_concat) error evaluating strings
2020-11-02 17:57:29	stdout	ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_card_driver returned error: No such file or directory
2020-11-02 17:57:29	stdout	ALSA lib confmisc.c:767:(parse_card) cannot find card '0'
2020-11-02 17:57:29	stdout	qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.80000007H:EDX.invtsc [bit 8]
2020-11-02 17:57:29	stdout	qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.01H:ECX.avx [bit 28]
2020-11-02 17:57:29	stdout	qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.01H:ECX.pcid [bit 17]
2020-11-02 17:57:29	stdout	qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.80000007H:EDX.invtsc [bit 8]
2020-11-02 17:57:29	stdout	qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.01H:ECX.avx [bit 28]
2020-11-02 17:57:29	stdout	qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.01H:ECX.pcid [bit 17]
2020-11-02 17:57:29	stdout	qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.80000007H:EDX.invtsc [bit 8]
2020-11-02 17:57:29	stdout	qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.01H:ECX.avx [bit 28]
2020-11-02 17:57:29	stdout	qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.01H:ECX.pcid [bit 17]
2020-11-02 17:57:29	stdout	qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.80000007H:EDX.invtsc [bit 8]
2020-11-02 17:57:29	stdout	qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.01H:ECX.avx [bit 28]
2020-11-02 17:57:29	stdout	(qemu) qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.01H:ECX.pcid [bit 17]
2020-11-02 17:57:29	stdout	QEMU 5.1.0 monitor - type 'help' for more information
2020-11-02 17:57:29	stdout	qemu-system-x86_64: falling back to tcg
2020-11-02 17:57:29	stdout	qemu-system-x86_64: failed to initialize kvm: No such file or directory
2020-11-02 17:57:29	stdout	Could not access KVM kernel module: No such file or directory
2020-11-02 17:57:28	stdout	Unable to init server: Could not connect: Connection refused
2020-11-02 17:57:26	stdout	nohup: appending output to 'nohup.out'
2020-11-02 17:57:26	stdout	ssh-keygen: generating new host keys: RSA DSA ECDSA ED25519 

```

**Comment by sickcodes** [comment]
I'm not that familiar with using Docker-OSX in NAS setups as I've never done one. It would be good to be able to test this though as a lot of users here have NAS arrangements.

**Comment by MdeLv** [comment]
I'm ready to make so me tests.
I don't see yet what needs to be done and where to start with.
I know that some NAS server are using a linux kernel, on which they install docker, as well as a virtual machine management GUI.  
Which plan do have you in mind?

What doesn't work:     Host (NAS server) -> Debian -> Docker (Arch btw) -> OSX
What doesn't work:     Host (NAS server) -> docker app -> Docker (Arch btw) -> OSX
What could be tested: Host (NAS server) -> docker app ->"adjusted Docker container with OSX"

What do you thinks about Synology DSM docker logs when using Docker-OSX?
Can you explain the architecture of 'Docker (Arch btw)' ? How it's connected to its host (docker) and to its guest ('OSX') ?
How could we adjust what is in Docker-OSX?

**Comment by sickcodes** [comment]
Docker runs an Arch Linux base image that uses the host X display server.

Any applications you run inside the Arch (btw) container, will be launched on your Desktop.

If you can't access the desktop running the Docker, then use the VNC version.

One of the applications I mentioned is QEMU.

Networking is done by exposing ports on the Arch container.

And inside the Arch container, it forwards the same ports from the OSX QEMU guest all the way to the host.

**Comment by sickcodes** [comment]
Most of these errors have been fixed and added to the https://github.com/sickcodes/Docker-OSX#troubleshooting

**Comment by MdeLv** [comment]
Good news!
Regarding using this docker image on debian or on a debian-based NAS, I understand that it's just about port mapping. So it's supposed to work more or less out of the box, isn't it ?

**Comment by MdeLv** [comment]
Just tried on a synology. 
I got a bunch of errors!
I see "kvm version too old" and so on.
As there is a web UI, will GTK be blocked?

Can you help to figure out how to fix that?
Thanks

```
sickcodes-docker-osx1
date,stream,content
2021-03-21 22:32:48,stdout,gtk initialization failed
2021-03-21 22:32:48,stdout,audio: Failed to create voice `adc'
2021-03-21 22:32:48,stdout,alsa: Reason: No such file or directory
2021-03-21 22:32:48,stdout,alsa: Failed to open `default':
2021-03-21 22:32:48,stdout,alsa: Could not initialize ADC
2021-03-21 22:32:48,stdout,ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM default
2021-03-21 22:32:48,stdout,ALSA lib conf.c:5233:(snd_config_expand) Evaluate error: No such file or directory
2021-03-21 22:32:48,stdout,ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
2021-03-21 22:32:48,stdout,ALSA lib confmisc.c:1246:(snd_func_refer) error evaluating name
2021-03-21 22:32:48,stdout,ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory
2021-03-21 22:32:48,stdout,ALSA lib confmisc.c:392:(snd_func_concat) error evaluating strings
2021-03-21 22:32:48,stdout,ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_card_driver returned error: No such file or directory
2021-03-21 22:32:48,stdout,ALSA lib confmisc.c:767:(parse_card) cannot find card '0'
2021-03-21 22:32:48,stdout,alsa: Reason: No such file or directory
2021-03-21 22:32:48,stdout,alsa: Failed to open `default':
2021-03-21 22:32:48,stdout,alsa: Could not initialize ADC
2021-03-21 22:32:48,stdout,ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM default
2021-03-21 22:32:48,stdout,ALSA lib conf.c:5233:(snd_config_expand) Evaluate error: No such file or directory
2021-03-21 22:32:48,stdout,ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
2021-03-21 22:32:48,stdout,ALSA lib confmisc.c:1246:(snd_func_refer) error evaluating name
2021-03-21 22:32:48,stdout,ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory
2021-03-21 22:32:48,stdout,ALSA lib confmisc.c:392:(snd_func_concat) error evaluating strings
2021-03-21 22:32:48,stdout,ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_card_driver returned error: No such file or directory
2021-03-21 22:32:48,stdout,ALSA lib confmisc.c:767:(parse_card) cannot find card '0'
2021-03-21 22:32:48,stdout,audio: Failed to create voice `dac'
2021-03-21 22:32:48,stdout,alsa: Reason: No such file or directory
2021-03-21 22:32:48,stdout,alsa: Failed to open `default':
2021-03-21 22:32:48,stdout,alsa: Could not initialize DAC
2021-03-21 22:32:48,stdout,ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM default
2021-03-21 22:32:48,stdout,ALSA lib conf.c:5233:(snd_config_expand) Evaluate error: No such file or directory
2021-03-21 22:32:48,stdout,ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
2021-03-21 22:32:48,stdout,ALSA lib confmisc.c:1246:(snd_func_refer) error evaluating name
2021-03-21 22:32:48,stdout,ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory
2021-03-21 22:32:48,stdout,ALSA lib confmisc.c:392:(snd_func_concat) error evaluating strings
2021-03-21 22:32:48,stdout,ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_card_driver returned error: No such file or directory
2021-03-21 22:32:48,stdout,ALSA lib confmisc.c:767:(parse_card) cannot find card '0'
2021-03-21 22:32:48,stdout,alsa: Reason: No such file or directory
2021-03-21 22:32:48,stdout,alsa: Failed to open `default':
2021-03-21 22:32:48,stdout,alsa: Could not initialize DAC
2021-03-21 22:32:48,stdout,ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM default
2021-03-21 22:32:48,stdout,ALSA lib conf.c:5233:(snd_config_expand) Evaluate error: No such file or directory
2021-03-21 22:32:48,stdout,ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
2021-03-21 22:32:48,stdout,ALSA lib confmisc.c:1246:(snd_func_refer) error evaluating name
2021-03-21 22:32:48,stdout,ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory
2021-03-21 22:32:48,stdout,ALSA lib confmisc.c:392:(snd_func_concat) error evaluating strings
2021-03-21 22:32:48,stdout,ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_card_driver returned error: No such file or directory
2021-03-21 22:32:48,stdout,ALSA lib confmisc.c:767:(parse_card) cannot find card '0'
2021-03-21 22:32:48,stdout,qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.80000007H:EDX.invtsc [bit 8]
2021-03-21 22:32:48,stdout,qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.01H:ECX.avx [bit 28]
2021-03-21 22:32:48,stdout,qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.01H:ECX.pcid [bit 17]
2021-03-21 22:32:48,stdout,qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.80000007H:EDX.invtsc [bit 8]
2021-03-21 22:32:48,stdout,qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.01H:ECX.avx [bit 28]
2021-03-21 22:32:48,stdout,qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.01H:ECX.pcid [bit 17]
2021-03-21 22:32:48,stdout,qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.80000007H:EDX.invtsc [bit 8]
2021-03-21 22:32:48,stdout,qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.01H:ECX.avx [bit 28]
2021-03-21 22:32:48,stdout,qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.01H:ECX.pcid [bit 17]
2021-03-21 22:32:48,stdout,qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.80000007H:EDX.invtsc [bit 8]
2021-03-21 22:32:48,stdout,qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.01H:ECX.avx [bit 28]
2021-03-21 22:32:48,stdout,(qemu) qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.01H:ECX.pcid [bit 17]
2021-03-21 22:32:48,stdout,QEMU 5.2.0 monitor - type 'help' for more information
2021-03-21 22:32:48,stdout,qemu-system-x86_64: falling back to tcg
2021-03-21 22:32:48,stdout,qemu-system-x86_64: failed to initialize kvm: Inappropriate ioctl for device
2021-03-21 22:32:47,stdout,kvm version too old
2021-03-21 22:32:47,stdout,Unable to init server: Could not connect: Connection refused
2021-03-21 22:32:46,stdout,nohup: appending output to 'nohup.out'
2021-03-21 22:32:46,stdout,chown: cannot access '': No such file or directory
2021-03-21 22:32:46,stdout,touch: cannot touch '': No such file or directory
```

**Comment by sickcodes** [comment]
Can you update KVM?

Also might be required:

`echo 1 | sudo tee /sys/module/kvm/parameters/ignore_msrs`


Do you want a display?

If so, you will need an X server running and use the `$DISPLAY` of that session to send the visual GTK QEMU there :)

**Comment by MdeLv** [comment]
As it's a synology server (so-called "NAS"), there seems to be a disappointing vendor lockin or "layering" (Vs a good free debian!)
From the DSM (DSM stands for DiskStation Manager : the main host on the HW server) :
```
$ sudo kvm --version
sudo: kvm: command not found
```
kvm should be installed for having a DSM (and vDSM, i.e. a virtual DSM), but it seems not to be available to the main host (the DSM). 
I don't know if KVM can be made available to Docker on a synology. Or maybe synology has added its own layer, or somehow tweaked debian/KVM...

It's unclear for me how docker (container manager) and KVM (VM hypervisor) are combined:
- in a normal debian : should work in parallel (docker or KVM, and docker provides KVM support to Docker-OSX container)
- in a synology "NAS"/server: ? can docker provide a container with an access to KVM??

According to you, what question should be asked to synology in the context of Docker-OSX? (host -> Arch -> MacOS)

Yes, I want a (distant) display.
Synology DSM has a webUI, so to connect to this "MacOS instance" from a mac/linux/win machine, I expect to use VNC or (local or distant) X client .

The bunch of errors reported are gtk/stdout ones. How can I avoid that?
Basically, container persistence put aside, the following command should be enough:

```
docker run -it \
    --device /dev/kvm \
    -p 50922:10022 \
    -v /tmp/.X11-unix:/tmp/.X11-unix \
    -e "DISPLAY=${DISPLAY:-:0.0}" \
    sickcodes/docker-osx:latest
```

What you call "headless Docker-OSX" is when using **no screen at all** Correct ?
https://github.com/sickcodes/Docker-OSX#how-to-run-in-headless-mode

**Comment by MdeLv** [comment]
FYI, synology has confirmed that there is **no** KVM support for their docker package (??!!.. wtf...).
I think that you'll confirm that KVM support is mandatory for Docker-OSX, w/o display... Correct?
Maybe installing a genuine docker package should do the job, if no vendor lockin also prevents me from doing that.
Or, do you have any idea of a solution to turn around that sad limitation?

**Comment by sickcodes** [comment]
I do, I'm testing the speed without KVM. I booted it without KVM on a raspberry Pi so this should work

https://github.com/sickcodes/Docker-OSX/pull/221

Untested yet

**Comment by MdeLv** [comment]
Any update?
Thanks.
